tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/RabbitMQ/PopulateBasicPropertiesHeadersIntegration.cs

```diff
@@ -45,8 +53,35 @@ public class PopulateBasicPropertiesHeadersIntegration
             return new CallTargetReturn<TReturn?>(returnValue);
         }
 
-        returnValue ??= CachedBasicPropertiesHelper<TReturn>.CreateHeaders();
-        var duckType = returnValue.DuckCast<IBasicProperties>()!;
+        TReturn? basicProperties = default;
+
+        // PopulateBasicPropertiesHeaders returns null if the supplied IReadOnlyBasicProperties
+        // does not have to be modified or if it's a writable instance.
+        // If that is the case then we have to fetch IReadOnlyBasicProperties from the argument
+        // list instead of creating a new instance that overwrites the supplied properties.
+        if (returnValue is null)
+        {
+            var rabbitMqClientAssembly = typeof(TReturn).Assembly;
+
+            if (state.State?.GetType() == rabbitMqClientAssembly.GetType("RabbitMQ.Client.BasicProperties", throwOnError: false))
+            {
+                // Use the existing basic properties if possible...
+                basicProperties = (TReturn)state.State!;
+            }
```

Comment:
I think we can simplify this a lot? 🤔 

```suggestion
            if (state.State is TReturn writable)
            {
                // Use the existing basic properties if possible...
                basicProperties = writable;
            }
```

Reply:
Oh yeah, `TReturn` is `RabbitMQ.Client.BasicProperties`. Thanks!